allennlp/modules/conditional_random_field_lannoy.py

```diff
@@ -0,0 +1,486 @@
+"""
+Conditional random field
+"""
+from typing import List, Tuple, Dict, Union
+
+import torch
+
+from allennlp.common.checks import ConfigurationError
+import allennlp.nn.util as util
+
+VITERBI_DECODING = Tuple[List[int], float]  # a list of tags, and a viterbi score
+
+
+def allowed_transitions(constraint_type: str, labels: Dict[int, str]) -> List[Tuple[int, int]]:
```

Comment:
Is this any different from the same function in `conditional_random_field.py`?

Reply:
I simply forgot to adapt this class. I think now it is much better.